src/VirtualClient/VirtualClient.Actions/SysbenchOLTP/SysbenchOLTPClientExecutor.cs

```diff
@@ -85,45 +87,76 @@ public string Workload
         /// <summary>
         /// The workload option passed to Sysbench.
         /// </summary>
-        public string NumTables
+        public int NumTables
         {
             get
             {
-                return this.Parameters.GetValue<string>(nameof(SysbenchOLTPClientExecutor.NumTables), 10);
+                int numTables = 10;
+
+                if (this.Parameters.TryGetValue(nameof(this.NumTables), out IConvertible value) && value != null
+                    && this.CpuArchitecture != Architecture.Arm64)
+                {
+                    numTables = value.ToInt32(CultureInfo.InvariantCulture);
+                }
+
+                return numTables;
             }
         }
 
         /// <summary>
         /// Number of records per table.
         /// </summary>
-        public string RecordCount
+        public int RecordCount
         {
             get
             {
-                this.Parameters.TryGetValue(nameof(SysbenchOLTPClientExecutor.RecordCount), out IConvertible recordCount);
-                return recordCount?.ToString();
+                // default formulaic setup of the database
+                // records & threads depend on the core count
+
+                int coreCount = Environment.ProcessorCount;
```

Comment:
I recommend getting it using Get CpuInfo so that it could be mocked and tested. You are doing log and power calculation and I'd like to see that unit tested